pkg/trace/api/api.go

```diff
@@ -309,17 +309,23 @@ func (r *HTTPReceiver) handleWithVersion(v Version, f func(Version, http.Respons
 	}
 }
 
+var (
+	errHeaderTraceCountNotFound = fmt.Errorf("%q header is not found", headerTraceCount)
+	errHeaderTraceCountNotSet   = fmt.Errorf("%q header value must be set", headerTraceCount)
+	errInvalidHeaderTraceCount  = fmt.Errorf("%q header value must be an integer", headerTraceCount)
+)
+
 func traceCount(req *http.Request) (int64, error) {
```

Comment:
I think you misunderstood me. We now have way too many errors. We just need one single error.
```go
var errInvalidTraceCountValue = fmt.Errorf("%q header value is not a number", headerTraceCount)
```
That's it.

Then, we can rewrite this function:
```go
func traceCount(req *http.Request) (int64, error) {
    if _, ok := req.Header[headerTraceCount]; !ok {
        return 0, fmt.Errorf("HTTP header %q not found", headerTraceCount)
    }
	n, err := strconv.Atoi(str)
	if err != nil {
		return 0, errInvalidTraceCountValue
	}
	return int64(n), nil
}
```
That's all. We don't actually need to check if it's empty. That was overkill. `strconv.Atoi` will fail on empty as well, and both outcomes mean the same thing: a number was not passed.

Then, we only log a message if `errInvalidTraceCountValue`. We log nothing otherwise.

Reply:
@gbbr 
Thank you for the detailed explanation.
https://github.com/DataDog/datadog-agent/pull/11804/commits/b48caa79fb410dd344c507bef85032adefa8b015 would follow your comment.